tests/unit/test_bruteforce_multi.cpp

```diff
@@ -0,0 +1,1663 @@
+#include "gtest/gtest.h"
+#include "VecSim/vec_sim.h"
+#include "test_utils.h"
+#include "VecSim/utils/arr_cpp.h"
+#include "VecSim/algorithms/brute_force/brute_force_multi.h"
+#include <cmath>
+
+class BruteForceMultiTest : public ::testing::Test {
+protected:
+    BruteForceMultiTest() {}
+
+    ~BruteForceMultiTest() override {}
+
+    void SetUp() override {}
+
+    void TearDown() override {}
+};
+
+TEST_F(BruteForceMultiTest, brute_force_vector_add_multiple_test) {
+    size_t dim = 4;
+    int rep = 5;
+    VecSimParams params{.algo = VecSimAlgo_BF,
+                        .bfParams = BFParams{.type = VecSimType_FLOAT32,
+                                             .dim = dim,
+                                             .metric = VecSimMetric_IP,
+                                             .multi = true,
+                                             .initialCapacity = 200}};
+    VecSimIndex *index = VecSimIndex_New(&params);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 0);
+
+    // Adding multiple vectors under the same label
+    for (int j = 0; j < rep; j++) {
+        float a[dim];
+        for (size_t i = 0; i < dim; i++) {
+            a[i] = (float)i * j + i;
+        }
+        VecSimIndex_AddVector(index, (const void *)a, 46);
+    }
+
+    ASSERT_EQ(VecSimIndex_IndexSize(index), rep);
+    ASSERT_EQ(reinterpret_cast<BruteForceIndex_Multi *>(index)->indexLabelCount(), 1);
+
+    // Deleting the label. All the vectors should be deleted.
+    VecSimIndex_DeleteVector(index, 46);
+
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 0);
+    ASSERT_EQ(reinterpret_cast<BruteForceIndex_Multi *>(index)->indexLabelCount(), 0);
+
+    VecSimIndex_Free(index);
+}
+
+/**** resizing cases ****/
+
+TEST_F(BruteForceMultiTest, resizeNAlignIndex) {
```

Comment:
good test, please check also the number of actual blocks, not only meta data containers